misc/images/materialized/entrypoint.sh

```diff
@@ -88,13 +88,21 @@ if [ -n "${MZ_RESTART_ON_FAILURE:-}" ]; then
     for ((i = 0; i < ${MZ_RESTART_LIMIT:-9999999999}; i++)); do
         # Run `environmentd` inside of an `if` to avoid tripping `set -e`
         # behavior.
+        # WIP: What's up with these identical branches? Should I maybe roll my
+        # logic from below into here?
         if environmentd "$@"; then
             code=$?
         else
             code=$?
         fi
-        echo "environmentd exited (code: $code); restarting in 5s..." >&2
-        sleep 5
+
+        if [[ $code > 0 ]]; then
+            echo "environmentd exited (code: $code); restarting in 5s..." >&2
+            sleep 5
+        else
+            echo "environmentd exited gracefully (code: $code); not restarting" >&2
```

Comment:
$code is always 0 here, otherwise for negative return codes we should also restart

Reply:
will push a change that makes it we restart on negative return values, just to match what we had before, but might not be useful ... 🤷‍♂️